Makefile

```diff
@@ -11,8 +11,16 @@ build.examples:
 
 .PHONY: test
 test:
+	# First we test the main module because the iteration through the modules
+	# in the lines above is very inconvenient when the folder is ".".
 	go test -tags=proxytest $(shell go list ./... | grep -v e2e)
 
+	# Now we go through the examples.
+	@find . -name "go.mod" \
```

Comment:
This is important, once we turn the examples into modules, they don't show up in `go list ./...` anymore and hence we need to run them one by one.